File: src/length.java
import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.File;
import java.io.FileReader;
import java.io.FileWriter;
import java.io.IOException;
import java.util.HashMap;
import java.util.Map;


public class length {

	public static void main(String[] args) {
		readFile();
		for(int j=0;j<9;j++){
			result[j]=toMeter(data[j]);
			if(j>5){
				result[j]=operate(data[j]);
			}
			System.out.println("result "+result[j]);
		}
		writeFile();
	}
	
	public static double todouble(String s){
		return  Double.parseDouble(s);
	}
	
	public static double getNum(String num){
		return todouble(num.substring(0,num.indexOf(" ")));
	}
	
	public static String getUnit(String str){
		return str.substring(str.indexOf(" ")+1);
	}
	
	public static double toMeter(String str){
		if(getUnit(str).equals("miles")){		return getNum(str) * todouble(myMap.get("mile"));	}
		if(getUnit(str).equals("yard")){		return getNum(str) * todouble(myMap.get("yard"));  	} 
		if(getUnit(str).equals("inches")){		return getNum(str) * todouble(myMap.get("inch"));  	} 
		if(getUnit(str).equals("feet")){		return getNum(str) * todouble(myMap.get("foot"));  	} 
		if(getUnit(str).equals("faths")){		return getNum(str) * todouble(myMap.get("fath"));  	} 
		if(getUnit(str).equals("furlong")){	return getNum(str) * todouble(myMap.get("furlong"));   } 
		return 0.0;
	}
	
	public static double operate(String str){
		int t=0;
		double re=0.0;
		while(str!=null){
			if(str.indexOf("+")!=-1){
				re=+toMeter(str.substring(0, str.indexOf("+")-1));
				str=str.substring(str.indexOf("+")+1);
			}else if(str.indexOf("-")!=-1){
				re=-toMeter(str.substring(0, str.indexOf("-")-1));
				str=str.substring(str.indexOf("-")+1);
			}else{
				str=null;
			}
		}
		return re;
	} 
	
	public static void readFile(){
		File file=new File("input.txt");
		
		if(!file.exists()){ System.out.println("no file"); return; }
	
		try {
            BufferedReader in = new BufferedReader(new FileReader(file));
            int i=0;
            while ((str = in.readLine()) != null) {
                if((temp++)<6){
                	myMap.put(str.substring(2,str.indexOf("=")-1),str.substring(str.indexOf("=")+2,str.length()-2));
                	
                }else if(str.length()!=0){
                	data[i++]=str;
				}       	
            }
            in.close();
        }catch (IOException e) {
            e.getStackTrace();
        }
	}

	public static void writeFile(){
		File file=new File("output.txt");
		
		if(!file.exists()){ System.out.println("no write file"); return; }
		
		try {
            BufferedWriter out = new BufferedWriter(new FileWriter(file));

            out.write("w812609862@gmail.com");
            out.newLine();
            
            for(int n=0;n<10;n++){
            	out.write(new java.text.DecimalFormat("#.00").format(result[n]));
            }
            out.close();
        }catch (IOException e) {
            e.getStackTrace();
        }
	}
	private static Map<String, String> myMap=new HashMap();
	private static int temp;
	private static String str;
	private static String[] data=new String[10];
	private static double result[]=new double[10];
	
}

File: length.cpp
#include <string>
#include <map>
#include <cstdio>
#include <iostream>
#include <vector>
using namespace std;



int pos,len,beg,ed;
string str,str_var;
map<string, double> dict;
map<string, string> fs;
double d,ans;


void init()
{
  fs["mile"] = "miles";
	fs["yard"] = "yards";
	fs["inch"] = "inches";
	fs["foot"] = "feet";
	fs["fath"] = "faths";
	fs["furlong"] = "furlongs";
}

void parse_string(string &str)
{
	
	vector<string> fh,num,_dw;
	ans = 0;

	str.push_back(' ');
	len = str.size();
	for(beg = ed = 0; ed < len; ed++)
	{
		if(str[ed] == ' ')
		{
			if(str[ed-1] == '+' || str[ed-1] == '-')
			{
				if(str[ed-1] == '+')
					fh.push_back("+");
				else
					fh.push_back("-");
			}else
			{
				if(str[ed-1] >= 'a' && str[ed-1] <= 'z')
				{
						_dw.push_back(str.substr(beg, ed-beg));
				}else
				{
					num.push_back(str.substr(beg, ed-beg));
				}
			}
			beg = ed+1;
		}
		
	}


	d = atof(&num[0][0]);
	d = d*dict[_dw[0]];
	ans += d;

	for(beg = 1; beg < num.size(); beg++)
	{
		d = atof(&num[beg][0]);
		d = d*dict[_dw[beg]];
		if(fh[beg-1][0] == '+')
			ans += d;
		else
			ans -= d;
	}
}





int main()
{	
	freopen("input.txt","r",stdin);
	freopen("output.txt","w",stdout);

	init();
	while(1)
	{	
		getline(cin, str);
		if(!str.size())
			break;
		len = str.size();
		pos = str.find('=');
		str_var = str.substr(pos+2, len-2-(pos+2)+1);
		d = atof(&str_var[0]);
		 str_var = str.substr(2, pos-2-2+1);
		dict[str_var] = d;
		dict[fs[str_var]] = d;
		str.clear();
	}

	printf("175119606@qq.com\n");
	while(1)
	{
		getline(cin, str);
		if(!str.size())
			break;
		printf("\n");
		parse_string(str);
		printf("%.2lf m",ans);
		str.clear();
	}
	return 0;
}
